#!/usr/bin/env python3
PACKAGE = "reu_lane_follow_pkg"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("hue_l", double_t, 0, "hue low",  0,   0, 179)
gen.add("hue_h", double_t, 0, "hue high", 179, 0, 179)

gen.add("sat_l", double_t, 0, "sat low",  0,   0, 255)
gen.add("sat_h", double_t, 0, "sat high", 255, 0, 255)

gen.add("val_l", double_t, 0, "val low",  230,   0, 255)
gen.add("val_h", double_t, 0, "val high", 255, 0, 255)

gen.add("lower_thresh", int_t, 0, "lower threshold",  200, 0, 254)
gen.add("upper_thresh", int_t, 0, "upper threshold", 255, 0, 255)
gen.add("aperture", int_t, 0, "aperture", 0, 0, 2)

gen.add("rho", int_t, 0, "rho value", 2, 1, 20)
gen.add("line_thresh", int_t, 0, "line threshold", 24, 1, 50)
gen.add("line_length", int_t, 0, "min line length", 40, 1, 70)
gen.add("line_gap", int_t, 0, "max line gap", 20, 1, 70)

exit(gen.generate(PACKAGE, "lane_blob_detect", "LaneThresh"))
